tests/test_ipu_dataloader.py

```diff
@@ -177,127 +187,26 @@ def test_poptorch_graphium_deviceiterations_gradient_accumulation(self):
-                features, labels = self.squeeze_input_dims(features, labels)
-                dict_input = {"features": features, "labels": labels}
-                preds = self.forward(dict_input)["preds"]
-                loss = self.compute_loss(
-                    preds, dict_input["labels"], weights=None, loss_fun=self.loss_fun, target_nan_mask=None
-                )
-                loss = self.pop.identity_loss(loss[0], reduction="mean")
-                return loss
-
-            def assert_shapes(self, dict_input, step):
-                msg = f", step=`{step}`"
-
-                # Test the shape of the labels
-                this_shape = list(dict_input["labels"]["homo"].shape)
-                true_shape = [self.batch_size, 2]
-                assert (
-                    this_shape == true_shape
-                ), f"Shape of the labels is `{this_shape}` but should be {true_shape}. {msg}"
-
-                # Test the shape of the node feature
-                this_shape = list(dict_input["features"]["feat"].shape)
-                true_shape = [self.node_batch_size, self.in_dims["feat"]]
-                assert (
-                    this_shape == true_shape
-                ), f"Shape of the feature 0 is `{this_shape}` but should be {true_shape}. {msg}"
-
-                # Test the shape of the node feature
-                this_shape = list(dict_input["features"]["edge_feat"].shape)
-                true_shape = [self.edge_batch_size, self.in_dims["edge_feat"]]
-                assert (
-                    this_shape == true_shape
-                ), f"Shape of the feature 0 is `{this_shape}` but should be {true_shape}. {msg}"
-
-            def get_progress_bar_dict(self):
-                return {}
-
-            def configure_optimizers(self):
-                return torch.optim.Adam(self.parameters(), lr=1e-3)
-
-            def squeeze_input_dims(self, features, labels):
-                for key, tensor in features:
-                    if isinstance(tensor, torch.Tensor):
-                        features[key] = features[key].squeeze(0)
-
-                for key in labels:
-                    labels[key] = labels[key].squeeze(0)
-
-                return features, labels
-
-        gradient_accumulation = 3
-        device_iterations = 5
-        batch_size = 7
-
-        # Initialize the batch info and poptorch options
-        opts = poptorch.Options()
-        opts.deviceIterations(device_iterations)
-        training_opts = deepcopy(opts)
-        training_opts.Training.gradientAccumulation(gradient_accumulation)
-        inference_opts = deepcopy(opts)
-
-        # Load the configuration file for the model
-        CONFIG_FILE = "tests/config_test_ipu_dataloader.yaml"
+        # Simplified testing config - reflecting the toymix requirements
+        # CONFIG_FILE = "tests/config_test_ipu_dataloader.yaml"
+        CONFIG_FILE = "tests/config_test_ipu_dataloader_multitask.yaml"
```

Comment:
Usually it's best to use the pytest fixture instead of absolute paths: https://github.com/datamol-io/graphium/blob/5397d66e31ec44a656e74ca7f3517f2e074d3679/tests/conftest.py#L10

This is minor, feel free to ignore.